src/panels/lovelace/elements/hui-group-element.ts

```diff
@@ -0,0 +1,151 @@
+import { createStyledHuiElement } from "../cards/picture-elements/create-styled-hui-element";
+
+import { LovelaceElement, LovelaceElementConfig } from "./types";
+import { HomeAssistant } from "../../../types";
+
+interface Config extends LovelaceElementConfig {
+  visible?: boolean;
+  toggle_tap?: boolean;
+  auto_scale?: boolean;
+  elements: LovelaceElementConfig[];
+}
+
+class HuiGroupElement extends HTMLElement implements LovelaceElement {
+  public _hass?: HomeAssistant;
+  private _config?: Config;
+  private _elements: LovelaceElement[] = [];
+  private _visible: boolean = true;
+  private _autoScale: boolean = false;
+
+  public setConfig(config: Config): void {
+    if (!config.elements || !Array.isArray(config.elements)) {
+      throw new Error("Error in card configuration.");
+    }
+
+    if (this._elements.length > 0) {
+      this._elements.map((el: LovelaceElement) => {
+        if (el.parentElement) {
+          el.parentElement.removeChild(el);
+        }
+      });
+
+      this._elements = [];
+    }
+
+    this._config = config;
+
+    this.style.transform = "none";
+    if (config.visible !== undefined) {
+      this._visible = config.visible;
+    }
+    if (config.auto_scale !== undefined) {
+      this._autoScale = config.auto_scale;
+    }
+
+    this.updateElements();
+  }
+
+  public firstStyled() {
+    if (!this._config) {
+      return;
+    }
+
+    this._config!.elements.map((elementConfig: LovelaceElementConfig) => {
+      const element = createStyledHuiElement(elementConfig);
+
+      if (this._autoScale) {
+        // adjust element dimensions according to parent
+        if (this.style.width) {
```

Comment:
Is this forcing a redraw of the element? And if so, should you just fetch it once? 

And does this work ? what if a picture is loaded later?

Reply:
It's in firstStyled so only called once.
It won't be affected by pictures or anything async since the idea is to take the position + size requested by the user via styles and adjust it to the size of the parent group container.
This allows people to just take the existing elements they have, stick them in to a group and then not worry about setting the positions and sizes again (it's all in % usually so it changes when changing parents).

Here's a pic for illustration of why auto scale is useful:

![image](https://user-images.githubusercontent.com/37745463/54273596-e586cc80-458e-11e9-842a-30c0288ba8af.png)